test/f32-raddextexp.cc

```diff
@@ -2,902 +2,112 @@
-  }
+      // Compute reference results.
+      double sum_ref = 0.0f;
+      for (size_t i = 0; i < elements(); i++) {
+        sum_ref += exp(double(x[i]));
+      }
 
-  TEST(F32_RADDEXTEXP__AVX2_P5_U80, elements_lt_80) {
-    TEST_REQUIRES_X86_AVX2;
-    for (size_t elements = 1; elements < 80; elements++) {
-      RAddExtExpMicrokernelTester()
-        .elements(elements)
-        .Test(xnn_f32_raddextexp_ukernel__avx2_p5_u80);
-    }
-  }
+      // Call optimized micro-kernel.
+      float sum[2];
+      raddextexp(elements() * sizeof(float), x.data(), sum);
 
-  TEST(F32_RADDEXTEXP__AVX2_P5_U80, elements_gt_80) {
-    TEST_REQUIRES_X86_AVX2;
-    for (size_t elements = 81; elements < 160; elements++) {
-      RAddExtExpMicrokernelTester()
-        .elements(elements)
-        .Test(xnn_f32_raddextexp_ukernel__avx2_p5_u80);
+      // Verify results.
+      ASSERT_NEAR(sum_ref, exp2(double(sum[1])) * double(sum[0]), std::abs(sum_ref) * 1.0e-6)
+        << "elements = " << elements() << ", y:value = " << sum[0] << ", y:exponent = " << sum[1];
     }
   }
-#endif  // XNN_ARCH_X86 || XNN_ARCH_X86_64
 
+ private:
+  size_t elements_{1};
+  size_t iterations_{15};
+};
 
-#if XNN_ARCH_X86 || XNN_ARCH_X86_64
-  TEST(F32_RADDEXTEXP__AVX2_P5_U80_ACC2, elements_eq_80) {
-    TEST_REQUIRES_X86_AVX2;
-    RAddExtExpMicrokernelTester()
-      .elements(80)
-      .Test(xnn_f32_raddextexp_ukernel__avx2_p5_u80_acc2);
+#define XNN_TEST_RADDEXTEXP_ELEMENT_EQ(ukernel, arch_flags, element_tile, ...)                                         \
+  TEST(ukernel, element_eq)                                                                                            \
+  {                                                                                                                    \
+    RAddExtExpMicrokernelTester().elements(element_tile).Test(ukernel);                                                \
   }
-
-  TEST(F32_RADDEXTEXP__AVX2_P5_U80_ACC2, elements_div_80) {
-    TEST_REQUIRES_X86_AVX2;
-    for (size_t elements = 160; elements < 800; elements += 80) {
-      RAddExtExpMicrokernelTester()
-        .elements(elements)
-        .Test(xnn_f32_raddextexp_ukernel__avx2_p5_u80_acc2);
-    }
-  }
-
-  TEST(F32_RADDEXTEXP__AVX2_P5_U80_ACC2, elements_lt_80) {
-    TEST_REQUIRES_X86_AVX2;
-    for (size_t elements = 1; elements < 80; elements++) {
-      RAddExtExpMicrokernelTester()
-        .elements(elements)
-        .Test(xnn_f32_raddextexp_ukernel__avx2_p5_u80_acc2);
-    }
+#define XNN_TEST_RADDEXTEXP_ELEMENT_DIV(ukernel, arch_flags, element_tile, ...)                                        \
+  TEST(ukernel, element_gt)                                                                                            \
+  {                                                                                                                    \
+    for (size_t element_size = element_tile * 2; element_size < element_tile * 10; element_size += element_tile) {     \
```

Comment:
`TEST_REQUIRES_ARCH_FLAGS(arch_flags)`?

Reply:
Thanks for pointing out the missing TEST_REQUIRES_ARCH_FLAGS(arch_flags). The issue with the illegal instruction for AVX-512 has now been resolved and now the tests are passing.